guild-factory/src/config.rs

```diff
@@ -23,6 +23,25 @@ pub trait ConfigModule {
         config_mapper.set(config_address);
     }
 
+    #[only_owner]
+    #[endpoint(upgradeConfigSc)]
+    fn upgrade_config_sc(&self, source_contract_address: ManagedAddress) {
```

Comment:
Verify `source_contract_address` is valid.

Reply:
I don't think such a check is needed. If you try to upgrade with an invalid contract, the calling of the "upgrade" function will fail.